### ISSUE TITLE
only display user's public tracks in track dropdown

### ISSUE THREAD

**Issue by berlintam** [sounddrop/sounddrop#58]
I logged into soundcloud w/ my maedels w/ a microphone account & noticed that some of our private tracks were also displayed in the track selection dropdown. We should remove these - because even associating them to a place probably won't do anything since they're not public anyway.....

**Comment by berlintam** [comment]
@nynnejc if you get the client `client = SoundCloud.new(:access_token => session[:access_token_hash]["access_token"])`  and then from there call `client.get('/me/tracks')` you get a list of tracks for your user. In the json that's returned you can see `"sharing" => "public"` (see example below). If a user has a private track, sharing will be set to private. 

```
[{"kind"=>"track",
  "id"=>120767789,
  "created_at"=>"2013/11/18 20:43:57 +0000",
  "user_id"=>8242020,
  "duration"=>20638,
  "commentable"=>true,
  "state"=>"finished",
  "original_content_size"=>3627918,
  "last_modified"=>"2016/03/21 10:19:28 +0000",
  "sharing"=>"public",
  "tag_list"=>"waterfall travel outdoors water winter cold",
  "permalink"=>"sounds-seljalandsfoss",
  "streamable"=>true,
  "embeddable_by"=>"all",
  "downloadable"=>true,
  "purchase_url"=>nil,
  "label_id"=>nil,
  "purchase_title"=>nil,
  "genre"=>"iceland",
  "title"=>"Sounds - Seljalandsfoss waterfall in Iceland",
  "description"=>"",
  "label_name"=>"",
  "release"=>"",
  "track_type"=>"",
  "key_signature"=>"",
  "isrc"=>"",
  "video_url"=>nil,
  "bpm"=>nil,
  "release_year"=>nil,
  "release_month"=>nil,
  "release_day"=>nil,
  "original_format"=>"wav",
  "license"=>"cc-by-nc-sa",
  "uri"=>"https://api.soundcloud.com/tracks/120767789",
  "user"=> etc... blahblah blah....
```

I guess the sharing attribute is the information you would need to somehow grab a hold of in order to do this ticket.  Maybe you can do something like this & pop it in somewhere... (warning, this is untested pseudocode)

```
tracks_to_share = [ ]
client.get('/me/tracks').each do |tracks|
   tracks_to_share << tracks["sharing"] == 'public'
end
```

Does that make sense? Let me know if it's total nonsense & we can go over it next friday/over slack/some other time etc...

woot!!!

**Comment by berlintam** [comment]
@nynnejc just some more context here: I was playing around w/ the app & tried to make a sounddrop w/ a private track of mine & the api returns a 404 can't find error when you do that. So this is additionally important because we don't have access to private tracks via the api (but oddly enough the api will still show them to us....)

**Comment by berlintam** [comment]
closed via #95
